src/update_handler.c

```diff
@@ -56,6 +57,54 @@ static GOutputStream* open_slot_device(RaucSlot *slot, int *fd, GError **error)
 	return outstream;
 }
 
+static gboolean empty_slot(RaucSlot *slot, GError **error)
+{
+	GError *ierror = NULL;
+	GInputStream *instream;
+	GOutputStream *outstream = NULL;
+	GFile *zerosrc = g_file_new_for_path("/dev/zero");
```

Comment:
Using /dev/zero from C seems strange. We could just write a zeroed page from memory until the device is full.